haystack/nodes/prompt/invocation_layer/hugging_face.py

```diff
@@ -196,22 +211,22 @@ def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union
         """
         n_prompt_tokens = len(self.pipe.tokenizer.tokenize(prompt))
         n_answer_tokens = self.max_length
-        if (n_prompt_tokens + n_answer_tokens) <= self.pipe.tokenizer.model_max_length:
+        if (n_prompt_tokens + n_answer_tokens) <= self.model_max_length:
             return prompt
 
         logger.warning(
             "The prompt has been truncated from %s tokens to %s tokens so that the prompt length and "
             "answer length (%s tokens) fit within the max token limit (%s tokens). "
             "Shorten the prompt to prevent it from being cut off",
             n_prompt_tokens,
-            self.pipe.tokenizer.model_max_length - n_answer_tokens,
+            max(0, self.model_max_length - n_answer_tokens),
```

Comment:
Why the `max` here? Do we expect `self.model_max_length - n_answer_tokens` to be negative?

Reply:
Yes, we do, but in some weird potential cases when model_max_length is specified too low we'll get the negative value here as n_answer_tokens is 100 by default.  So then the warning will be something like `The prompt has been truncated from 26 tokens to -90 tokens so that the prompt...` 
26 here is an arbitrary length of some hypothetical short prompt, model_max_length is for example 10 and n_answer_tokens is default 100.